libosrmc/osrmc.cc

```diff
@@ -26,9 +26,32 @@ int osrmc_is_abi_compatible(void) { return osrmc_get_version() >> 16u == OSRMC_V
 /* API */
 
 struct osrmc_error final {
+  std::string code;
   std::string message;
 };
 
+static void osrmc_error_from_exception(const std::exception& e, osrmc_error_t* error) {
+  *error = new osrmc_error{"Exception", e.what()};
+}
+
+static void osrmc_error_from_json(osrm::json::Object* json, osrmc_error_t* error) try {
+  if (json == nullptr) {
```

Comment:
Looks like you cherry-picked the other pull rerquest's commit. Just make sure to update this branch when you make changes (such as asserting not null here).

Reply:
I've been rebasing this on top of #7, so should hopefully avoid causing any conflicts.